cli/src/pcluster/models/cluster_config.py

```diff
@@ -433,24 +433,26 @@ def __init__(self, enabled: bool, port: int = None, allowed_ips: str = None):
 class Efa(Resource):
     """Represent the EFA configuration."""
 
-    def __init__(self, enabled: bool = None, gdr_support: bool = None):
-        super().__init__()
-        self.enabled = Resource.init_param(enabled, default=True)
+    def __init__(self, enabled: bool = None, gdr_support: bool = None, **kwargs):
```

Comment:
We can extend `Efa` to accept instance_type and initialize the enabled value according to this param if passed.
```python
def __init__(self, enabled: bool = None, gdr_support: bool = None, instance_type: str = None, **kwargs):
        super().__init__(**kwargs)

        default_enabled = AWSApi.instance().ec2.get_instance_type_info(instance_type).is_efa_supported() if instance_type else True
        self.enabled = Resource.init_param(gdr_support, default=default_enabled)
        self.gdr_support = Resource.init_param(gdr_support, default=False)
```

then initializing it with:
```python
self.efa = efa or Efa(instance_type=self.instance_type, implied=True)
```

This should simplify tests and mocking.